prompt2model/utils/parse_responses.py

```diff
@@ -28,56 +31,18 @@ def find_and_parse_json(
         final response as a Dictionary
         Else returns None.
     """
-    response_text = response.choices[0]["message"]["content"]
-
-    # Find JSON objects in the response, by matching text between curly braces.
-    potential_jsons = re.findall(r"\{.*?\}", response_text, re.DOTALL)
-    for response_text in potential_jsons:
-        try:
-            response_json = json.loads(response_text, strict=False)
-        except json.decoder.JSONDecodeError:
-            logger.warning(f"API response was not a valid JSON: {response_text}")
-            continue
-
-        missing_keys = [key for key in required_keys if key not in response_json]
-        if len(missing_keys) != 0:
-            logger.warning(f'API response must contain {", ".join(required_keys)} keys')
-            continue
-
-        final_response = {}
-        for key in required_keys + optional_keys:
-            if key not in response_json:
-                # This is an optional key, so exclude it from the final response.
-                continue
-            if type(response_json[key]) == str:
-                final_response[key] = response_json[key].strip()
-            else:
-                final_response[key] = response_json[key]
-        return final_response
-    return None
-
+    if type(response) != str and hasattr(response, "choices"):
```

Comment:
You should usually not use `type(...) ==` in Python; use `isinstance` instead ([source](https://switowski.com/blog/type-vs-isinstance/)).
```suggestion
    if not isinstance(response, str) and hasattr(response, "choices"):
```